16/umbraco-cms/customizing/extending-overview/extension-types/global-context.md

```diff
@@ -1,24 +1,164 @@
+- **Share functionality with other extensions** - Allow other package developers to access features and functionality within the package
+
+Extension authors should prefer to use other context types [Workspace Contexts](workspaces/workspace-context.md) over Global Contexts. Umbraco itself uses Global Contexts sparingly, for clipboard, current user, and icons.
 
 ## Registration of a Global Context
 
-You can register a global context like so:
+Global Context extensions can be registered using a manifest file like `umbraco-package.json`.
+
+{% code title="umbraco-package.json" %}
+```json
+{
+    "$schema": "../../umbraco-package-schema.json",
+    "name": "My Global Context Package",
+    "version": "1.0.0",
+    "extensions": [
+        {
+            "type": "globalContext",
+            "alias": "My.GlobalContext",
+            "name": "My Global Context",
+            "api": "/App_Plugins/my-global-context/dist/my-context.context.js"
+        }
+    ]
+}
+```
+{% endcode %}
+
+**Key fields:**
+- `type`: Must be `"globalContext"`, see: [ManifestGlobalContext](https://apidocs.umbraco.com/v16/ui-api/interfaces/packages_core_extension-registry.ManifestGlobalContext.html)
+- `alias`: A unique identifier for the context
+- `name`: A human-readable name
+- `api`: The path to the compiled JavaScript file
+
+## Creating Your Global Context
+
+### 1. Define a Context Token
+
+First, create a context token. This is how other parts of an extension will reference the context:
+
+{% code title="src/my-context.ts" %}
+```typescript
+import { UmbContextToken } from '@umbraco-cms/backoffice/context-api';
+
+export interface MyGlobalContextInterface {
+    getCurrentUser(): Promise<string>;
+    setPreference(key: string, value: any): void;
+    getPreference(key: string): any;
+}
+
+export const MY_GLOBAL_CONTEXT = new UmbContextToken<MyGlobalContext>(
+    'GlobalContext'
+);
+```
+{% endcode %}
+
+{% hint style="info" %}
+When defining the context token, reference the context class instead of the interface for the generic type. Ex: `UmbContextToken<MyGlobalContext>`, instead of `UmbContextToken<MyGlobalContextInterface>`
+{% endhint %}%}
+
+{% hint style="info" %}
+The string passed into the contextAlias parameter does not need to match the alias registered in the manifest.
+{% endhint %}%}
+
+### 2. Implement the Context Class
+
+Next, implement the context class:
 
+{% code title="src/my-context.ts" %}
 ```typescript
+import { UmbContextBase } from '@umbraco-cms/backoffice/class-api';
+import { UmbControllerHost } from '@umbraco-cms/backoffice/controller-api';
+
+export class MyGlobalContext
+    extends UmbContextBase
+    implements MyGlobalContextInterface
 {
-    type: 'globalContext',
-    alias: 'My.GlobalContext.Products',
-    name: 'My Products Context',
-    api: 'my-products.context.js',
+    #preferences: Map<string, any> = new Map();
+
+    constructor(host: UmbControllerHost) {
+        super(host, SERVICE_STATUS_CONTEXT);
```

Comment:
`SERVICE_STATUS_CONTEXT` does not exist in this example.. 

It maye be an idea to just change the name of `MY_GLOBAL_CONTEXT` and its string value `GlobalContext`. :-)
to SERVICE_STATUS_CONTEXT, so the examples are cohrent.